modules/data_dictionary_widget/tests/src/Functional/DataDictionaryWidgetTest.php

```diff
@@ -53,6 +53,80 @@ public function testDataDictionaryWidgetBehavior() {
+    $page->fillField('field_json_metadata[0][dictionary_fields][field_collection][group][title]', $test_data_dictionary_field_title);
+    $page->selectFieldOption('field_json_metadata[0][dictionary_fields][field_collection][group][type]', $test_data_dictionary_field_type);
+    $page->selectFieldOption('field_json_metadata[0][dictionary_fields][field_collection][group][format]', $test_data_dictionary_field_format);
+    $page->fillField('field_json_metadata[0][dictionary_fields][field_collection][group][description]', $test_data_dictionary_field_description);
+    $page->pressButton('Add');
+
+    // Assert the values are being displayed.
+    $session->pageTextContains($test_data_dictionary_field_name);
+    $session->pageTextContains($test_data_dictionary_field_title);
+    $session->pageTextContains('Data Type: ' . $test_data_dictionary_field_type);
+    $session->pageTextContains('Format: ' . $test_data_dictionary_field_format);
+    $session->pageTextContains('Description: ' .  $test_data_dictionary_field_description);
+
+    // Edit data dictionary fields and confirm fields have the same values.
+    $page->pressButton('Edit');
+    $session->fieldValueEquals('field_json_metadata[0][dictionary_fields][data][0][field_collection][name]', $test_data_dictionary_field_name);
+    $session->fieldValueEquals('field_json_metadata[0][dictionary_fields][data][0][field_collection][title]', $test_data_dictionary_field_title);
+    $session->fieldValueEquals('field_json_metadata[0][dictionary_fields][data][0][field_collection][type]', $test_data_dictionary_field_type);
+    $session->fieldValueEquals('field_json_metadata[0][dictionary_fields][data][0][field_collection][format]', $test_data_dictionary_field_format);
+    $session->elementTextContains('css', 'textarea[name="field_json_metadata[0][dictionary_fields][data][0][field_collection][description]"]', $test_data_dictionary_field_description);
+    $page->pressButton('Cancel');
+
+    // Fill the indexes fields.
+    $page->pressButton('Add index');
+    $test_index_title = 'Test Index';
+    $test_index_type = 'index';
+    $test_index_name = 'test_index';
+    $test_index_length = 20;
+    $page->fillField('field_json_metadata[0][indexes][field_collection][group][index][description]', $test_index_title);
+    $page->selectFieldOption('field_json_metadata[0][indexes][field_collection][group][index][type]', $test_index_type);
+    // Add an index field.
+    // Use the name to distinguish the button from the data dictionary fields.
+    $page->pressButton('add_index_field');
+    $page->fillField('field_json_metadata[0][indexes][fields][field_collection][group][index][fields][name]', $test_index_name);
+    $page->fillField('field_json_metadata[0][indexes][fields][field_collection][group][index][fields][length]', $test_index_length);
+    // Need to distinguish the add button on the index fields vs the one for data dictionary fields.
+    $page->pressButton('Add ');
```

Comment:
Odd that this has a space after but good that it allows for a distinguishment between the two, we should probably make this more unique in later iterations, something like "add new index field".